.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,51 @@
+name: facebook/metro/build-and-test
+on:
+  pull_request:
+    types: [opened, synchronize]
+
+# head_ref is per PR, so this ensures that updating the latest PR commit
+# will cancel the previous run of the workflow and trigger a new one
+concurrency:
+  group: "build-and-test-${{ github.head_ref }}"
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash
```

Comment:
Q: Do we have any compatibility issues with zsh (assuming as default)? I'd imagine everything we'd need would work, since zsh is a superset of Bash.

Reply:
Unfortunately zsh is not installed on ubuntu images (and installing it is +20sec to each run)
We can use macOS machines instead, but even there zsh is not "officially supported" but only supported as a "custom shell" running a template: https://github.com/actions/runner-images/issues/264 
I still tested it to ensure it doesn't work out of the box, and it indeed doesn't:
https://github.com/facebook/metro/actions/runs/10286164783